### PR TITLE
gobblin salesforce update to new proxy

### PR DESCRIPTION
**Pull request by tuGithub**

### CODE REVIEW

**Comment by tuGithub** [comment]
1. changed multiple workUnit.getProp -> workUnitState.getProp
2. changed DEFAULT_USE_ALL_OBJECTS from true to false
3. backward compatible with 0.2.805 version of JST file while getting the low/high water mark

**Comment by tuGithub** [comment]
@chavdar , please review it when you have time

File: gobblin-core/src/main/java/gobblin/source/extractor/extract/QueryBasedSource.java [hunk 1]
```diff
@@ -224,16 +225,40 @@ public void shutdown(SourceState state) {}
...
+      
       String table = previousWus.getExtract().getTable();
 
-      long lowWm = previousWus.getWorkunit().getLowWatermark(LongWatermark.class).getValue();
+      long lowWm = ConfigurationKeys.DEFAULT_WATERMARK_VALUE;
+      LongWatermark waterMarkObj = previousWus.getWorkunit().getLowWatermark(LongWatermark.class);
+      // new job state file(version 0.2.1270) , water mark format:
+      // "watermark.interval.value": "{\"low.watermark.to.json\":{\"value\":20160101000000},\"expected.watermark.to.json\":{\"value\":20160715230234}}",
+      if(waterMarkObj != null){
+        lowWm = previousWus.getWorkunit().getLowWatermark(LongWatermark.class).getValue();
```
Comment (chavdar) [inline gobblin-core/src/main/java/gobblin/source/extractor/extract/QueryBasedSource.java]: nit: waterMarkObj.getValue()

File: gobblin-core/src/main/java/gobblin/source/extractor/extract/QueryBasedSource.java [hunk 2]
```diff
@@ -224,16 +225,40 @@ public void shutdown(SourceState state) {}
...
         prevLowWatermarksByTable.put(table, lowWm);
       } else {
         prevLowWatermarksByTable.put(table, Math.min(prevLowWatermarksByTable.get(table), lowWm));
       }
 
-      long highWm = previousWus.getActualHighWatermark(LongWatermark.class).getValue();
+      long highWm = ConfigurationKeys.DEFAULT_WATERMARK_VALUE;
+      waterMarkObj = previousWus.getActualHighWatermark(LongWatermark.class);
+      if(waterMarkObj != null){
+        highWm = previousWus.getActualHighWatermark(LongWatermark.class).getValue();
```
Comment (chavdar) [inline gobblin-core/src/main/java/gobblin/source/extractor/extract/QueryBasedSource.java]: nit: waterMakrObj.getValue()

**Comment by chavdar** [comment]
Could you explain "2. changed DEFAULT_USE_ALL_OBJECTS from true to false" ?

**Comment by tuGithub** [comment]
with DEFAULT_USE_ALL_OBJECTS set to true, the single salesforce  job will pull all the data

File: gobblin-core/src/main/java/gobblin/source/extractor/extract/QueryBasedSource.java (hunk 1 above)
Comment (tuGithub) [inline gobblin-core/src/main/java/gobblin/source/extractor/extract/QueryBasedSource.java]: fixed

File: gobblin-core/src/main/java/gobblin/source/extractor/extract/QueryBasedSource.java (hunk 2 above)
Comment (tuGithub) [inline gobblin-core/src/main/java/gobblin/source/extractor/extract/QueryBasedSource.java]: fixed

**Pull request merged**